scripts/dts/gen_defines.py

```diff
@@ -478,10 +478,13 @@ def write_children(node: edtlib.Node) -> None:
     out_dt_define(f"{node.z_path_id}_CHILD_NUM", len(node.children))
 
     ok_nodes_num = 0
-    for child in node.children.values():
+    for i, child in enumerate(node.children.values()):
         if child.status == "okay":
             ok_nodes_num = ok_nodes_num + 1
 
+        out_dt_define(f"{node.z_path_id}_CHILD_IDX_{i}", f"DT_{child.z_path_id}")
+        out_dt_define(f"{node.z_path_id}_CHILD_IDX_{i}_EXISTS", 1)
+
```

Comment:
even this code does not even try to be deterministic, I'm not even sure what will be the order of iteration.